src/arrayMethodSort.js

```diff
@@ -4,8 +4,34 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    let sortingFunc;
+
+    // Compare elements as strings if compareFunction is omitted
+    if (compareFunction !== undefined) {
+      sortingFunc = compareFunction;
+    } else {
+      sortingFunc = (a, b) => String(a) > String(b);
+    }
+
+    // Iterate through the array {this.length - 1} amount of times,
+    // skipping the sorted values
+    for (let maxLength = this.length; maxLength > 0; maxLength--) {
+      for (let i = 1; i < maxLength; i++) {
+        const a = this[i - 1];
+        const b = this[i];
+        const comparissonResult = sortingFunc(a, b);
+
+        // Swap objects positions if the result of comparison is positive
+        if (comparissonResult > 0) {
+          this[i - 1] = b;
+          this[i] = a;
+        }
+      }
+    }
```

Comment:
The current implementation of the bubble sort will work, but it's known to be inefficient for larger arrays. In a real-world scenario, you might want to consider using more efficient sorting algorithms like quicksort or mergesort for better performance.